### PR TITLE
Feature/auth role/rpeck

### PR DESCRIPTION
**Pull request by andromedis**
## Description

- onlyRoles middleware created which allows endpoints to be restricted to certain roles
- onlyRoles middleware contains exceptions for users to read and update their own data
- onlyRoles has been added to all endpoints only used in admin tools components

#### Context:

- Security vulnerability on the front end allows anyone to gain admin rights and access/edit/delete sensitive data
- Back end verification of user role for role-restricted endpoints fixes issue without changing front end
- Front end may also patch this in the future?

Fixes # Security Vulnerability

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have removed unnecessary comments/console logs from my code
- [ ] I have made corresponding changes to the documentation if necessary (optional)
- [X] My changes generate no new warnings
- [X] I have checked my code and corrected any misspellings
- [X] No duplicate code left within changed files
- [X] Size of pull request kept to a minimum
- [X] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

File: api/middleware/onlyRoles.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/**
+ * Accepts an array of role_ids which have permission to access the endpoint,
+ * if the role_id in the profile added to the request header by authRequired
+ * middleware matches one of the role_ids in the parameter array, proceeds with
+ * the request, otherwise sends a 403 response and denies access. It also allows
+ * any user to read or update their own data.
+ */
```
Comment (elCarlosSantiago) [inline api/middleware/onlyRoles.js:7]: Love this comment, will provide a lot of clarity for later devs!

**Review by elCarlosSantiago — APPROVED** [review]
Really great work Becca!

File: api/middleware/onlyRoles.js (hunk 1 above)
Comment (bharukimiller) [reply api/middleware/onlyRoles.js:7]: Seconded. Really good so people can easily understand what this is doing in the future. Maybe the only thing needed to be added is an explanation to using 1 and 2 hardcoded to represent our roles could be added, though it is easy to figure out after interfacing with the code.

**Review by bharukimiller — APPROVED** [review]
Good authorization. We may need to expand for some future features, but for now, this is great!

File: api/middleware/onlyRoles.js (hunk 1 above)
Comment (elCarlosSantiago) [reply api/middleware/onlyRoles.js:7]: Good point. I think we can keep it as is and include an update to the documentation: i.e: 

"""
Admin: 1
Moderator: 2
User: 3
"""

**Pull request merged**